### PR TITLE
Feature: chatgpt UI describer

### PR DESCRIPTION
**Pull request by antonieto**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,3 +4,4 @@ chrome-extension/node_modules
 .mypy_cache
 .ruff_cache
 __pycache__
+env
```
Comment (CarlosCRG19) [inline .gitignore:7]: nice.

Little suggestion, I normally name virtual envs as `.venv` which is already in this gitignore. This prevents it from being displayed when you run `ls` while avoiding collision with `.env` files

File: server/adapters/element_describer.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+class ChatGPTDescriber(ElementDescriber):
+    api_key: str
+    project: str
+    client: OpenAI
+
+    PROMPT = '''
+    The image you will receive is a cropped UI element from an application.
+    Your task is to describe the element in sufficent but concise detail.
+    You will limit your response to a maximum of 2.
```
Comment (CarlosCRG19) [inline server/adapters/element_describer.py:22]: a maximum of 2 what?

File: server/adapters/element_describer.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    You must absolutely include the following information in your response:
+        - The type of element (e.g. button, text field, etc.)
+        - The text content of the element (if any)
+        - The color of the element
+        - The shape of the element (if applicable)
+    
+    Make sure you incldue all of the above information in the THREE sentences you provide.
```
Comment (CarlosCRG19) [inline server/adapters/element_describer.py:29]: previously, maximum of 2 sentences (or at least thats what I guess) but here you ask for 3?

File: server/setup.sh [hunk 4]
```diff
@@ -0,0 +1,5 @@
+# Bash script to setup the environment for the project
+python3 -m venv env
+source ./env/bin/activate
+pip install -r requirements.txt
+
```
Comment (CarlosCRG19) [inline server/setup.sh:5]: nice

File: server/adapters/element_describer.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+                  },
+                },
+              ],
+            }
+          ],
+          max_tokens=300,
+        )
+
+        if not response.choices or not response.choices[0]:
+            raise DescriptionException('No description could be generated.')
```
Comment (CarlosCRG19) [inline server/adapters/element_describer.py:61]: nit, you can rewrite the class definition to include the message
```
class DescriptionException(Exception):
    def __init__(self, message="An error occurred with the description."):
        self.message = message
        super().__init__(self.message)
```

**Review by CarlosCRG19 — APPROVED** [review]

File: server/adapters/element_describer.py (hunk 3 above)
Comment (antonieto) [reply server/adapters/element_describer.py:31]: I updated the prompt: 
```
 The image you will receive is a cropped UI element from an application.
    Your task is to describe the element in sufficent but concise detail.
    You will limit your response to a maximum of 2 sentences.
    You must absolutely include the following information in your response:
        - The type of element (e.g. button, text field, etc.)
        - The text content of the element (if any)
        - The color of the element
        - The shape of the element (if applicable)

    Make sure you incldue all of the above.
    Limit your response to TWO sentences but remember this is a maximum.
    Whenever possible, use A SINGLE SENTENCE.
    Do not include irrelevant information such as:
        - The purpose of the element
        - The functionality of the element
        - The position of the element
    Only include visual details.
```

**Pull request closed without merging**